src/util/LogSlowExecution.cpp

```diff
@@ -44,36 +45,45 @@ LogSlowExecution::checkElapsedTime() const
     {
         std::lock_guard<std::mutex> guard(gLogSlowExecMutex);
 
+        // Check whether we're 10 times over the threshold to decide the log
+        // level.
+        LogLevel ll = elapsed > mThreshold * 10 ? LogLevel::LVL_WARNING
+                                                : LogLevel::LVL_DEBUG;
+
         // Only emit a new log message once-per-second (optionally preceded
         // by a summary of how many messages have been dropped since last time).
-        if ((finish - gLastLogMessage) > std::chrono::seconds(1))
+        if ((finish - gLastLogMessage[ll].first) > std::chrono::seconds(1))
```

Comment:
nit: just create a local ref to `gLastLogMessage[ll]` to avoid looking it up all the time/makes the code easier to follow

Reply:
I added a ref for both elements in the pair for readability!